src/zap.c

```diff
@@ -4074,10 +4074,13 @@ static const short elem_pairings[][3] = {
     { PM_WATER_ELEMENTAL, ZT_COLD, PM_ICE_ELEMENTAL },
     { PM_WATER_ELEMENTAL, ZT_ACID, PM_ACID_ELEMENTAL },
     { PM_EARTH_ELEMENTAL, ZT_FIRE, PM_MAGMA_ELEMENTAL },
+    { PM_EARTH_ELEMENTAL, ZT_ACID, PM_MUD_ELEMENTAL },
     { PM_FIRE_ELEMENTAL, ZT_LIGHTNING, PM_PLASMA_ELEMENTAL },
     /* { PM_AIR_ELEMENTAL, ZT_ELEC, PM_STORM_ELEMENTAL }, */
     { PM_ICE_ELEMENTAL, ZT_FIRE, PM_WATER_ELEMENTAL },
+    { PM_ICE_ELEMENTAL, ZT_SONIC, PM_WATER_ELEMENTAL },
     { PM_MAGMA_ELEMENTAL, ZT_COLD, PM_EARTH_ELEMENTAL },
+    { PM_MAGMA_ELEMENTAL, ZT_ACID, PM_FIRE_ELEMENTAL },
```

Comment:
These two don't make a ton of sense to me. Presumably, no matter how much you shatter an ice elemental with sonic attacks, it won't actually heat up enough to melt. (More of a snow elemental type of thing?)

Fire elementals are classically made of fire without any solid substance. Hitting a molten-rock being with acid doesn't seem like it would separate the heat from the rock.